src/services/teams/hooks/helpers.js

```diff
@@ -0,0 +1,481 @@
+
+        return { userId, role };
+    });
+};
+
+/**
+ * @helper
+ * @private
+ * @param {Array::teamUser} teamUsers 
+ * @param {Array||String} userIds 
+ * @return {Array::teamUser}
+ */
+const removeTeamUsers = (teamUsers, userIds) => {
+    if (isString(userIds))
+        userIds = [userIds];
+
+    if (!isArrayWithElement(userIds))
+        return teamUsers;
+
+    return teamUsers.reduce((list, teamUser) => {
+        if (!userIds.includes(teamUser.userId))
+            list.push(teamUser);
+        return list;
+    }, []);
+};
+
+/**
+ * Remove all elements if they are not in team schools.
+ * @helper
+ * @private
+ * @requires function::removeTeamUsers
+ * @param {Array::String::ObjectId::schools} schoolIds 
+ * @param {Array::TeamUser} teamUsers 
+ * @param {Array::User} users 
+ * @return {Array::TeamUser}
+ */
+const removeNotValidUsersBySchoolIds = (schoolIds, teamUsers, users) => {
+    let removeList = [];
+    schoolIds = bsonIdToString(schoolIds);
+
+    teamUsers.map(teamUser => {
+        const user = users.find(_user => isSameId(_user._id, teamUser.userId));
+        const schoolId = bsonIdToString(user.schoolId);
+        if (schoolIds.includes(schoolId) === false)
+            removeList.push(user._id);
+    });
+    return removeTeamUsers(teamUsers, removeList);
+};
+
+const teamOwnerRoleExist = (hook, teamUsers, oldTeam, users) => {
+    //todo later
+    // const teamownerRoleId = hook.findRole('name', 'teamowner', '_id');
+    // remove logic in mappedInputUserIdsToTeamUsers for owner 
+    // test if any of this users has the role teamowner 
+    // teamowner must have the userRole teacher!
+    return teamUsers;
+};
+
+/**
+ * Remove duplicated users. It test it over key>userId. Role of first found are used.
+ * @param {Array::TeamUser} teamUsers 
+ * @return {Array::TeamUser}
+ */
+const removeDublicatedTeamUsers = (teamUsers) => {
+    let foundId = [];
+    return teamUsers.reduce((stack, _teamUser) => {
+        const id = bsonIdToString(_teamUser.userId);
+        if (foundId.includes(id) === false) {
+            stack.push(_teamUser);
+            foundId.push(id);
+        }
+        return stack;
+    }, []);
+};
+
+
+/**
+ * @helper
+ * @requires function::mappedInputUserIdsToTeamUsers
+ * @requires function::removeNotValidUsersBySchoolIds
+ * @param {Object::hook} hook 
+ * @param {Array::Object::User} users 
+ * @return {Array::TeamUser, default:[]} 
+ */
+const getTeamUsers = (hook, team, users) => {
+    let teamUsers = hook.data.userIds;
+
+    if (isObject(teamUsers) || isString(teamUsers))
+        teamUsers = [teamUsers];
+
+    if (!isArrayWithElement(teamUsers))
+        return [];
+
+    teamUsers = mappedInputUserIdsToTeamUsers(hook, teamUsers, team);
+    teamUsers = teamOwnerRoleExist(hook, teamUsers, team, users);
+    teamUsers = removeDublicatedTeamUsers(teamUsers);
+    teamUsers = removeNotValidUsersBySchoolIds(team.schoolIds, teamUsers, users);
```

Comment:
I'm not sure if these functions are used elsewhere too, but these four lines look like they are all implementing filters, so it would be nicer to implement them as such (returning a boolean for a single user) and then just use `teamUsers.filter(user => foo(user) && bar(user) && baz(user))`

Reply:
Only two of them are real filter. The other are complex operation with data transformation. For handle it as single operations in my opinion it is easier to understand.